### PR TITLE
Refactor testGetWithTimeout for Reliability

### PR DESCRIPTION
**Pull request by arturobernalg**
- Use AtomicBoolean for future completion tracking.
- Simplify async behavior and improve test determinism.

### CODE REVIEW

File: httpcore5/src/test/java/org/apache/hc/core5/concurrent/TestBasicFuture.java [hunk 1]
```diff
@@ -260,35 +260,36 @@ public void cancelled() {
...
-                future.completed("test");
-            } catch (final InterruptedException e) {
-                future.failed(e);
-            }
-        }).start();
+        new Thread(() -> future.completed("test")).start();
 
+        // Poll until the future is completed or timeout
         assertTimeoutPreemptively(Duration.ofMillis(200), () -> {
+            while (!isFutureCompleted.get()) {
```
Comment (ok2c) [inline httpcore5/src/test/java/org/apache/hc/core5/concurrent/TestBasicFuture.java:288]: @arturobernalg Is busy loop the only way to reliably test this condition?

Comment (arturobernalg) [reply httpcore5/src/test/java/org/apache/hc/core5/concurrent/TestBasicFuture.java:288]: @ok2c 
A busy loop (or spinning loop) is not the only way to reliably test asynchronous conditions. In fact, in many cases, it's not the most efficient or recommended approach due to its CPU-intensive nature

if you believe the busy loop approach is more suitable for our testing environment, we can implement it.

**Review by ok2c — APPROVED** [review]

File: httpcore5/src/test/java/org/apache/hc/core5/concurrent/TestBasicFuture.java (hunk 1 above)
Comment (thc202) [reply httpcore5/src/test/java/org/apache/hc/core5/concurrent/TestBasicFuture.java:288]: Wouldn't a `CountDownLatch` work for this case?

Comment (arturobernalg) [reply httpcore5/src/test/java/org/apache/hc/core5/concurrent/TestBasicFuture.java:288]: Yes, a `CountDownLatch` is indeed an alternative to achieve the same result.

Comment (thc202) [reply httpcore5/src/test/java/org/apache/hc/core5/concurrent/TestBasicFuture.java:288]: It's not an alternative.

**Pull request merged**